src/power_grid_model/__init__.py

```diff
@@ -6,6 +6,7 @@
 
 from power_grid_model.core.power_grid_meta import initialize_array, power_grid_meta_data
 from power_grid_model.core.power_grid_model import PowerGridModel
+from power_grid_model.dataset_definitions import ComponentType, DataType
```

Comment:
These names are too similar. 
Lets have the type hints (ie. `ComponentType`) in the data_types.py along with `SingleDataset` etc. 
Or maybe a separate file `typing.py` or something for the distinction.

It feels out of place in `__init__.py`. Here we can only keep the enum, `ComponentTypes`.

Reply:
@nitbharambe Per my recent investigation, `Literal` does not really provide advantages in this scenario since dict[Literal, Any] is just interpreted as dict[str, Any] by Python, hence my new solution to this.

@mgovers In the end we will probably end up With `DataType` containing the `Enum` definition.